Use Python SOABI

This uses the Python sysconfig to find the proper extension suffix for
the Python module that is built.

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Upstream-status: Pending

Index: git/src/qt_gui_cpp_sip/CMakeLists.txt
===================================================================
--- git.orig/src/qt_gui_cpp_sip/CMakeLists.txt
+++ git/src/qt_gui_cpp_sip/CMakeLists.txt
@@ -46,12 +46,17 @@ if(sip_helper_FOUND)
     DEPENDENCIES qt_gui_cpp
   )
 
+  execute_process(
+    COMMAND ${Python3_EXECUTABLE} -c "import sysconfig as c; print(c.get_config_var('EXT_SUFFIX'), end='')"
+    OUTPUT_VARIABLE PYTHON_EXTENSION_MODULE_SUFFIX
+    ERROR_QUIET)
+
   if(APPLE)
     set(LIBQT_GUI_CPP_SIP_SUFFIX .so)
   elseif(WIN32)
     set(LIBQT_GUI_CPP_SIP_SUFFIX .pyd)
   else()
-    set(LIBQT_GUI_CPP_SIP_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(LIBQT_GUI_CPP_SIP_SUFFIX ${PYTHON_EXTENSION_MODULE_SUFFIX})
   endif()
 
   install(FILES ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_PYTHON_DESTINATION}/libqt_gui_cpp_sip${LIBQT_GUI_CPP_SIP_SUFFIX}
